### ISSUE TITLE
Change Environment Inside Flutter App after runApp()

### ISSUE THREAD

**Issue by davidmednikov** [java-james/flutter_dotenv#21]
I have this plugin loading the environment in `main.dart`, as follows:

```
await DotEnv().load('env/stage.env'); // Stage Environment
// await DotEnv().load('env/prod.env'); // Prod Environment
runApp(MyApp());
```

I am able to comment / uncomment the environment I want when debugging and building the app, and this has been working very well for us.

However, we got a request from a user to be able to switch environments while running the app. I have the trigger to do this hidden to admins only, however after the user selects a new environment, I am unable to actually reload DotEnv to use a different environment. When I call `await DotEnv().load()` with the newly selected environment, nothing is actually changed.

I'm assuming that this is due to the plugin using a singleton that is initialized before the actual app widget is run, but perhaps I am just missing something. Is there any way to change the loaded environment after the initial load?

Thanks in advance to the dev for an incredible plugin!

**Comment by java-james** [comment]
Thanks @davidmednikov for raising the issue. Although intended to be used in initialisation and swapped out with the desired `.env` on bundle, I would expect `_env` to mutate if load was called again. (I foresee keys that don't match could be an issue here as they would be retained during the mutation due to using `addAll`)

I haven't looked at the code in a while so I will take a look.

Can you confirm it isn't due to the containing widget not updating state to represent the new env

**Comment by java-james** [comment]
Have published 3.0.0. This also works with reloading the environment. 2.0.0 also supports clearing the environment before loading a new one.

**Comment by java-james** [comment]
Closing due to inability to reproduce - Give 3.0.0 a spin if the issue persists for you please reopen this issue :)

**Comment by davidmednikov** [comment]
Thank you for the update! Will give 3.0.0 a shot and post here with any updates.

I can confirm that the widget state has updated after the load because I call `setState(() {})` after reloading the environment.

Regarding the assets: I have the assets added as such:

`assets:
   - env/`

I will try listing them individually if updating to 3.0.0 doesn't work right off the bat.

**Comment by davidmednikov** [comment]
I don't think that Github allows me to re-open the issue, but still not having success after updating to version 3.0.0 and explicitly importing my assets. Not sure why it's not working for me but maybe I'm doing something wrong? It's difficult to share my code because it is proprietary, but I will do my best.

file `pubspec.yaml`:

```
assets:
   - env/prod.env
   - env/stage.env
```

file `stage.env`:

```
ENVIRONMENT=STAGE
```

file `prod.env`:

```
ENVIRONMENT=PROD
```

file `environment_enum.dart`:
```
enum EnvironmentEnum {
  dev,
  stage,
  prod,
}
```

file `environment_service.dart`:

```
import 'package:flutter_dotenv/flutter_dotenv.dart' as DotEnv;
...
static Future<void> loadEnvironment({EnvironmentEnum environment = EnvironmentEnum.prod}) async {
    await DotEnv.load(fileName: 'env/${EnumToString.convertToString(environment)}.env');
}
...
static final String _environment = DotEnv.env['ENVIRONMENT'];
...
static EnvironmentEnum get environment => EnumToString.fromString(EnvironmentEnum.values, _environment);
```

I am using the following code to reload the environment from stage to prod:

```
print(environment);
await loadEnvironment(); // defaults to PROD environment
print(environment);
```

The output is:

```
I/flutter (14574): STAGE
I/flutter (14574): STAGE
```

I would expect the second print statement to output `PROD` to the console.

I've tried making the properties non-static but that didn't seem to fix it.

Any ideas?

**Comment by java-james** [comment]
I will reopen issue and look into this.

**Comment by java-james** [comment]
Before I do can you please confirm it isn't because your _environment is declared outside of mutable state and mutable widget state as a `final`. 
I can't be 100% with the code shown but suspecting that's a potential issue for you. I wouldn't expect _environment to update here unless mutable and set again.

Eg. try `var _environment = DotEnv.env['ENVIRONMENT'];`

And then set environment again after DotEnv.load
eg. 
```
static Future<void> loadEnvironment({EnvironmentEnum environment = EnvironmentEnum.prod}) async {
    await DotEnv.load(fileName: 'env/${EnumToString.convertToString(environment)}.env');
    _environment = DotEnv.env['ENVIRONMENT'];
}
```

**Comment by davidmednikov** [comment]
Thank you!! Your last suggestion helped me figure out my problem. I was still having the same issue after removing `final` from the variable declaration, as the getter was still trying to access a variable that had already been set. Then I removed the variable assignments to only use getters, similar to what you suggested. So now the getters are reading directly from the `DotEnv` instance, just like this:

```
static String get environmentString => DotEnv.env['ENVIRONMENT'];
static EnvironmentEnum get environment => EnumToString.fromString(EnvironmentEnum.values, DotEnv.env['ENVIRONMENT'])
```

For reference, to make the fix work throughout the entire app, I had to replace all assignments that referenced the `EnvironmentService` with getters. For example, in a different file, I replaced the following:

```
final dioClient = Dio(endpoint);

static BaseOptions endpoint = BaseOptions(
    baseUrl: EnvironmentService.endpoint,
);
```

with:

```
Dio get dioClient => Dio(endpoint );

BaseOptions get endpoint => BaseOptions(
    baseUrl: EnvironmentService.endpoint,
);
```

Everything is working perfectly now. Thank you again!
